src/textual/worker.py

```diff
@@ -306,22 +306,20 @@ def run_callable(work: Callable[[], ResultType]) -> ResultType:
             active_worker.set(self)
             return work()
 
+        loop = asyncio.get_running_loop()
         if (
             inspect.iscoroutinefunction(self._work)
             or hasattr(self._work, "func")
             and inspect.iscoroutinefunction(self._work.func)
         ):
-            runner = run_coroutine
-        elif inspect.isawaitable(self._work):
-            runner = run_awaitable
-        elif callable(self._work):
-            runner = run_callable
-        else:
-            raise WorkerError("Unsupported attempt to run a thread worker")
+            return await loop.run_in_executor(None, run_coroutine, self._work)  # type: ignore[arg-type]
```

Comment:
Is this really better? Can't we modify the original in a simpler way to satisfy Mypy?

Reply:
The type for the `runner` local is very complicated and would need to be explicitly stated here